torchrl/envs/utils.py

```diff
@@ -181,34 +187,46 @@ def step_mdp(
             next_tensordict.update(out)
             return next_tensordict
         return out
-    out = tensordict.get("next").clone(False)
-    excluded = set()
-    if exclude_done:
-        excluded.add(done_key)
-    if exclude_reward:
-        excluded.add(reward_key)
-    if len(excluded):
-        out = out.exclude(*excluded, inplace=True)
-    td_keys = None
+
+    td = tensordict.exclude("next")
+    td_next = tensordict.get("next")
+
+    td_keys = td.keys(True, True)
+    td_next_keys = td_next.keys(True, True)
+
+    out = TensorDict({}, td_next.batch_size)
+
+    # Set the keys from next
+    excluded = {
+        done_key if exclude_done else None,
+        reward_key if exclude_reward else None,
+    }
```

Comment:
Isn't weird to have None in the set?
Should we not build it iteratively?

Reply:
i can do it, i thought preformance was similar and it was more readable, but i ll change it